admin/config-ui/class-configuration-endpoint.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * @package WPSEO\Admin\ConfigurationUI
+ */
+
+/**
+ * Class WPSEO_Configuration_Endpoint
+ */
+class WPSEO_Configuration_Endpoint {
+
+	const REST_NAMESPACE = 'yoast/v1';
+	const ENDPOINT_RETRIEVE = 'configurator';
+	const ENDPOINT_STORE = 'configurator';
+
+	const CAPABILITY_RETRIEVE = 'manage_options';
+	const CAPABILITY_STORE = 'manage_options';
+
+
+	/** @var WPSEO_Configuration_Service Service to use */
+	protected $service;
+
+	/**
+	 * @param WPSEO_Configuration_Service $service Service to use.
+	 */
+	public function set_service( WPSEO_Configuration_Service $service ) {
+		$this->service = $service;
+	}
+
+	/**
+	 * Register REST routes.
+	 */
+	public function register() {
+		// Register fetch config.
+		register_rest_route( self::REST_NAMESPACE, self::ENDPOINT_RETRIEVE, array(
+			'methods'             => 'GET',
+			'callback'            => array(
+				$this,
+				'get_configuration',
+			),
+			'permission_callback' => array(
+				$this,
+				'can_retrieve_data',
+			),
+		) );
+
+		// Register save changes.
+		register_rest_route( self::REST_NAMESPACE, self::ENDPOINT_STORE, array(
+			'methods'             => 'PUT',
+			'callback'            => array(
+				$this->service,
+				'set_configuration',
+			),
+			'permission_callback' => array(
+				$this,
+				'can_save_data',
+			),
+		) );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_configuration() {
```

Comment:
This method is more like a wrapper, we might consider removing this method and call the `get_configuration` of `this->service` instead.